hazelcast/src/main/java/com/hazelcast/map/impl/LatencyTrackingMapLoader.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl;
+
+import com.hazelcast.core.MapLoader;
+import com.hazelcast.internal.diagnostics.StoreLatencyPlugin;
+import com.hazelcast.internal.diagnostics.StoreLatencyPlugin.LatencyProbe;
+
+import java.util.Collection;
+import java.util.Map;
+
+public class LatencyTrackingMapLoader<K, V> implements MapLoader<K, V> {
+
+    static final String KEY = "MapStoreLatency";
```

Comment:
I thought about this. However..  if a user would implement a MapStore,then part of its information is in the MapStoreLatency subtree and the other part in the MapLoaderLatency section.

So the information tree above.